language/stdlib/modules/DesignatedDealer.move

```diff
@@ -93,6 +93,22 @@ module DesignatedDealer {
             add_currency<CoinType>(dd, tc_account);
         };
     }
+    spec fun publish_designated_dealer_credential {
+        pragma opaque = true;
+        let dd_addr = Signer::spec_address_of(dd);
+        modifies global<Dealer>(dd_addr);
+        ensures exists<Dealer>(dd_addr);
+        modifies global<TierInfo<CoinType>>(dd_addr), global<TierInfo<Coin1>>(dd_addr), global<TierInfo<Coin2>>(dd_addr);
+        ensures add_all_currencies ==> exists<TierInfo<Coin1>>(dd_addr) && exists<TierInfo<Coin2>>(dd_addr);
+        ensures !add_all_currencies ==> exists<TierInfo<CoinType>>(dd_addr);
+
+        include Roles::AbortsIfNotTreasuryCompliance{account: tc_account};
+        include Roles::AbortsIfNotDesignatedDealer{account: dd};
+        aborts_if exists<Dealer>(dd_addr) with Errors::ALREADY_PUBLISHED;
+        include !add_all_currencies ==> AddCurrencyAbortsIf<CoinType>{dd_addr: dd_addr};
```

Comment:
You can also do `include if (add_all_currences) AddCurrency... else AddCurrency... && AddCurrency...;`

Reply:
Done.